yt_dlp/extractor/mastodon.py

```diff
@@ -0,0 +1,515 @@
+            'url': str_or_none(dict_get(media, ('url', 'source_mp4'))),
+            'thumbnail': str_or_none(media['preview_url']) if media['type'] == 'video' else None,
+            'vcodec': 'none' if media['type'] == 'audio' else None,
+            'duration': float_or_none(try_get(media, lambda x: x['meta']['original']['duration'])),
+            'width': int_or_none(try_get(media, lambda x: x['meta']['original']['width'])),
+            'height': int_or_none(try_get(media, lambda x: x['meta']['original']['height'])),
+            'tbr': int_or_none(try_get(media, lambda x: x['meta']['original']['bitrate'])),
+        } for media in metadata.get('media_attachments') or [] if media.get('type') in ('video', 'audio')]
+
+        if not entries:
+            card = metadata.get('card')
+            if not card:
+                raise ExtractorError('No audio/video attachments', expected=True)
+            return {
+                '_type': 'url_transparent',
+                'url': card['url'],
+                'title': title,
+                'thumbnail': url_or_none(card.get('image')),
+                **info_dict,
+            }
+
+        if len(entries) == 1:
+            del entries[0]['id']
+            info_dict.update(entries[0])
+        else:
+            info_dict.update({
+                '_type': 'playlist',
+                'entries': entries,
+            })
+
+        return info_dict
+
+
+class MastodonInstancesIE(MastodonBaseIE):
+    _BASE_IES = (MastodonIE, )
+    _WEBPAGE_TESTS = [{
+        'note': 'embed video without NSFW',
+        'url': 'https://mstdn.jp/@nao20010128nao/105395495018076252',
+        'info_dict': {
+            'id': '105395495018076252@mstdn.jp',
+            'ext': 'mp4',
+            'like_count': int,
+            'comment_count': int,
+            'uploader': 'Lesmiscore',
+            'repost_count': int,
+            'uploader_id': 'nao20010128nao',
+            'title': 'てすや\nhttps://www.youtube.com/watch?v=jx0fBBkaF1w',
+            'uploader_url': 'https://mstdn.jp/@nao20010128nao',
+            'thumbnail': 'https://media.mstdn.jp/media_attachments/files/033/830/003/small/e8429d6ee1013c3e.png',
+            'duration': 131.598,
+        },
+    }, {
+        'note': 'embed video with NSFW',
+        'url': 'https://mstdn.jp/@nao20010128nao/105395503690401921',
+        'info_dict': {
+            'id': '105395503690401921@mstdn.jp',
+            'ext': 'mp4',
+            'comment_count': int,
+            'uploader_id': 'nao20010128nao',
+            'uploader_url': 'https://mstdn.jp/@nao20010128nao',
+            'like_count': int,
+            'repost_count': int,
+            'thumbnail': 'https://media.mstdn.jp/media_attachments/files/033/830/052/small/3c6bdd3e3aa4ed65.png',
+            'uploader': 'Lesmiscore',
+            'duration': 131.598,
+            'title': 'Mastodonダウンローダーのテストケース用なので別に注意要素無いよ',
+            'age_limit': 18,  # TODO
```

Comment:
@Lesmiscore

Reply:
```diff
$ git diff
diff --git a/yt_dlp/extractor/mastodon/mastodon.py b/yt_dlp/extractor/mastodon/mastodon.py
index 8ee5d7946..e2d36a070 100644
--- a/yt_dlp/extractor/mastodon/mastodon.py
+++ b/yt_dlp/extractor/mastodon/mastodon.py
@@ -210,7 +210,6 @@ def suitable(cls, url):
             'title': 'てすや\nhttps://www.youtube.com/watch?v=jx0fBBkaF1w',
             'uploader': 'nao20010128nao',
             'uploader_id': 'nao20010128nao',
-            'age_limit': 0,
         },
     }, {
         'note': 'embed video with NSFW',
@@ -259,7 +258,6 @@ def suitable(cls, url):
             'title': 'てすや\nhttps://www.youtube.com/watch?v=jx0fBBkaF1w',
             'uploader': 'nao20010128nao',
             'uploader_id': 'nao20010128nao',
-            'age_limit': 0,
         },
     }, {
         # mastodon, video description
@@ -424,6 +422,8 @@ def _real_extract(self, url):
             'uploader': traverse_obj(metadata, ('account', 'display_name')),
             'uploader_id': traverse_obj(metadata, ('account', 'username')),
             'uploader_url': traverse_obj(metadata, ('account', 'url')),
+
+            'age_limit': 18 if metadata.get('sensitive') else None
         }
 
         if domain == 'gab.com':
```
I thought there were extraction for `age_limit`